### PR TITLE
Pa 535 trigger snapshot process with nsb extensions

### PR DESCRIPTION
**Pull request by Pauljgraham**

### CODE REVIEW

File: src/SFA.DAS.NServiceBus.AzureFunction.Extensions.UnitTests/WhenLoggingIncomingBehaviour.cs [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+using NServiceBus.Testing;
+using NServiceBus.Unicast.Messages;
+using NUnit.Framework;
+using SFA.DAS.NServiceBus.AzureFunction.Extensions.UnitTests.Data;
+
+namespace SFA.DAS.NServiceBus.AzureFunction.Extensions.UnitTests;
+public class WhenLoggingIncomingBehaviour
+{
+    [Test]
+    public async Task Should_not_fail_when_called()
```
Comment (benagain) [inline src/SFA.DAS.NServiceBus.AzureFunction.Extensions.UnitTests/WhenLoggingIncomingBehaviour.cs:12]: I'm not sure there's an awful lot of value to tests that merely assert that an entire class "doesn't throw an exception."  If we can't test the behaviour of `LogIncomingBehaviour` (and I don't think we want to) then I would simply remove these tests.

File: src/SFA.DAS.NServiceBus.Extensions.UnitTests/WhenSettingUpEndpointConfiguration.cs [hunk 2]
```diff
@@ -0,0 +1,22 @@
+﻿using NServiceBus;
+using NUnit.Framework;
+
+
+namespace SFA.DAS.NServiceBus.Extensions.UnitTests;
+
+public class WhenSettingUpEndpointConfiguration
+{
+    [Test]
+    public void Then_UseSerializer_does_not_throw_error()
```
Comment (benagain) [inline src/SFA.DAS.NServiceBus.Extensions.UnitTests/WhenSettingUpEndpointConfiguration.cs:10]: Again, I think either test actual behaviour or remove the tests.  These tests don't add any value IMO.

File: src/SFA.DAS.TrackProgress.Jobs.v3.ncrunchsolution [hunk 3]
```diff
@@ -0,0 +1,6 @@
+﻿<SolutionConfiguration>
```
Comment (benagain) [inline src/SFA.DAS.TrackProgress.Jobs.v3.ncrunchsolution]: Do we normally include the ncruch files in the repository or add them to `.gitignore`?

**Review by benagain — COMMENTED** [review]
Good idea to include a test event publisher

File: src/SFA.DAS.NServiceBus.AzureFunction.Extensions.UnitTests/WhenLoggingIncomingBehaviour.cs (hunk 1 above)
Comment (Pauljgraham) [reply src/SFA.DAS.NServiceBus.AzureFunction.Extensions.UnitTests/WhenLoggingIncomingBehaviour.cs:12]: To be resolved elsewhere

File: src/SFA.DAS.TrackProgress.Jobs.v3.ncrunchsolution (hunk 3 above)
Comment (Pauljgraham) [reply src/SFA.DAS.TrackProgress.Jobs.v3.ncrunchsolution]: good point, ncrunch should be ignored

**Pull request closed without merging**